core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java

```diff
@@ -147,25 +148,27 @@ public synchronized OutputStream createCompressionStream(OutputStream downStream
     },
 
     GZ(COMPRESSION_GZ) {
-      private transient DefaultCodec codec;
+      private transient AtomicReference<DefaultCodec> codec = new AtomicReference<DefaultCodec>();
 
       @Override
-      synchronized CompressionCodec getCodec() {
-        if (codec == null) {
-          codec = new DefaultCodec();
-          codec.setConf(conf);
+      CompressionCodec getCodec() {
+        DefaultCodec resultCodec = codec.get();
+        if (null == resultCodec) {
+          DefaultCodec newCodec = new DefaultCodec();
```

Comment:
The merit of creating it at construction time depends on how costly the initialization is. I'm under the impression that it's not _that_ bad, so eager initialization is probably ok here and we don't have to worry about concurrency at all. Good point, @brianloss.